tests/Annotation/EntityTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Opstalent\CrudBundle\Tests\Annotation;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Opstalent\CrudBundle\Annotation\Entity;
+use Opstalent\CrudBundle\Tests\AnnotationTestCase;
+use Opstalent\CrudBundle\Tests\ObjectManipulator;
+use Opstalent\CrudBundle\Tests\Resources\Entity\AllAvailableActionsAnnotatedModel;
+
+/**
+ * Class EntityTest
+ * @author Szyon Kunowski <szymon.kunowski@gmail.com>
+ * @package Opstalent\CrudBundle
+ */
+class EntityTest extends AnnotationTestCase
+{
+    /**
+     * @return null|object
+     */
+    protected function getMetadata()
+    {
+        $reader = new AnnotationReader();
+        return $reader->getClassAnnotation(
+            new \ReflectionClass(AllAvailableActionsAnnotatedModel::class),
+            Entity::class
+        );
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function isAnnotation()
+    {
+        $this->assertIsAnnotation(Entity::class);
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function entityReturnProperActionForGivenOptions()
+    {
+        $actions = ['addable', 'getable', 'listable', 'editable', 'deletable'];
+        $reader = new AnnotationReader();
+        $entity = new Entity(['actions' => $actions]);
+        $this->assertEquals($actions, $entity->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function entityReturnAllCorrectActionForGivenOptions()
+    {
+        $actions = ['addable', 'getable', 'listable', 'editable', 'deletable', 'adorable'];
+        $reader = new AnnotationReader();
+        $entity = new Entity(['actions' => $actions]);
+        $this->assertNotEquals($actions, $entity->getActions());
+        $properActions = ['addable', 'getable', 'listable', 'editable', 'deletable'];
+        $this->assertEquals($properActions, $entity->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     **/
+    public function entityAnnotationIsUsed()
```

Comment:
isn't it an integration test? also it's not necessary

Reply:
fixed